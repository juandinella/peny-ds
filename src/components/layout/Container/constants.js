export const options = {
  /**
   * Justify Content
   * https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content
   */
  justifyContent: [
    'flex-start',
    'center',
    'flex-end',
    'space-between',
    'space-around',
    'space-evenly',
    'stretch',
  ],
  /**
   * Align Items
   * https://developer.mozilla.org/en-US/docs/Web/CSS/align-items
   */
  alignItems: ['flex-start', 'center', 'flex-end', 'stretch', 'baseline'],
  /**
   * Align Content
   * https://developer.mozilla.org/en-US/docs/Web/CSS/align-content
   */
  alignContent: [
    'stretch',
    'center',
    'flex-start',
    'flex-end',
    'space-between',
    'space-around',
    'stretch',
  ],
  /**
   * Flex DIrection
   * https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction
   */
  direction: ['row', 'row-reverse', 'column', 'column-reverse'],
  /**
   * Flex Wrap
   * https://developer.mozilla.org/en-US/docs/Web/CSS/flex-wrap
   */
  wrap: ['nowrap', 'wrap', 'wrap-reverse'],
  spaces: ['xxs', 'xs', 'sm', 'md', 'lg', 'xl', 'xxl', 'xxxl'],
  borderRadiuses: ['xs', 'sm', 'lg'],
  borderColors: [
    'neutral-900',
    'neutral-800',
    'neutral-700',
    'neutral-600',
    'neutral-500',
    'neutral-400',
    'neutral-300',
    'neutral-200',
    'neutral-100',
    'neutral-50',
    'neutral-00',
    'primary-main',
    'secondary-main',
  ],
  overflow: ['hidden', 'visible', 'scroll', 'auto'],
}
