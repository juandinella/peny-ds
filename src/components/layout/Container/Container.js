import React from 'react'
import PropTypes from 'prop-types'
import styles from './Container.styles'
import { options } from './constants'

/**
 * Container
 */

const Container = ({
  id,
  space,
  mspace,
  maxWidth,
  minWidth,
  gap,
  mgap,
  flex,
  borderColor,
  borderVertical,
  borderHorizontal,
  borderRadius,
  justifyContent,
  mJustifyContent,
  alignItems,
  mAlignItems,
  alignContent,
  mAlignContent,
  direction,
  mDirection,
  wrap,
  children,
  className,
  centered,
  leftAligned,
  rightAligned,
  isVisible,
  fullWidthMobile,
  maxContent,
  negativeMargin,
  minContent,
  hideMobile,
  hideTablet,
  showTablet,
  showMobile,
  blockTablet,
  overflow,
  absolute,
  top,
  right,
  bottom,
  left,
  fullHeight,
}) => {
  return (
    <div
      className={className}
      id={id}
      data-testid={id}
      css={styles}
      style={{ top: top, right: right, bottom: bottom, left: left }}
      maxWidth={maxWidth}
      minWidth={minWidth}
      gap={gap}
      mgap={mgap}
      space={space}
      mspace={mspace}
      borderColor={borderColor}
      borderVertical={borderVertical}
      borderHorizontal={borderHorizontal}
      borderRadius={borderRadius}
      flex={flex}
      alignItems={alignItems}
      mAlignItems={mAlignItems}
      justifyContent={justifyContent}
      mJustifyContent={mJustifyContent}
      alignContent={alignContent}
      mAlignContent={mAlignContent}
      direction={direction}
      mDirection={mDirection}
      wrap={wrap}
      centered={centered}
      leftAligned={leftAligned}
      rightAligned={rightAligned}
      isVisible={isVisible}
      fullWidthMobile={fullWidthMobile}
      maxContent={maxContent}
      minContent={minContent}
      hideMobile={hideMobile}
      negativeMargin={negativeMargin}
      showTablet={showTablet}
      showMobile={showMobile}
      hideTablet={hideTablet}
      blockTablet={blockTablet}
      overflow={overflow}
      absolute={absolute}
      fullHeight={fullHeight}
    >
      {children}
    </div>
  )
}

Container.displayName = 'Container'

Container.propTypes = {
  /** Container's id */
  id: PropTypes.string,
  /** Container's children */
  children: PropTypes.node.isRequired,
  /** Container's Padding */
  space: PropTypes.oneOf(options.spaces),
  /** Container's Padding for mobile*/
  mspace: PropTypes.oneOf(options.spaces),
  /** Container's Border Radius */
  borderRadius: PropTypes.oneOf(options.borderRadiuses),
  /** Container's Border Color */
  borderColor: PropTypes.oneOf(options.borderColors),
  /** Set maxWidth of the container */
  maxWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The space Between containers */
  gap: PropTypes.string,
  /** The space Between containers for mobile*/
  mgap: PropTypes.string,
  /** Set minWidth of the container */
  minWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The CSS align-items property sets the align-self value on all direct children as a group. In Flexbox, it controls the alignment of items on the Cross Axis. */
  alignItems: PropTypes.oneOf(options.alignItems),
  /** Align Items form mobile */
  mAlignItems: PropTypes.oneOf(options.alignItems),
  /** The CSS justify-content property defines how the browser distributes space between and around content items along the main-axis of a flex container. */
  justifyContent: PropTypes.oneOf(options.justifyContent),
  /** Justify Content form mobile */
  mJustifyContent: PropTypes.oneOf(options.justifyContent),
  /** The CSS align-content property sets the distribution of space between and around content items along a flexbox's cross-axis */
  alignContent: PropTypes.oneOf(options.alignContent),
  /** Align Content for mobile */
  mAlignContent: PropTypes.oneOf(options.alignContent),
  /** The flex-direction CSS property sets how flex items are placed in the flex container defining the main axis and the direction (normal or reversed). */
  direction: PropTypes.oneOf(options.direction),
  /** Flex direction for mobile */
  mDirection: PropTypes.oneOf(options.direction),
  /** The flex-wrap CSS property sets whether flex items are forced onto one line or can wrap onto multiple lines. If wrapping is allowed, it sets the direction that lines are stacked. */
  wrap: PropTypes.oneOf(options.wrap),
  /** Container's overflow */
  overflow: PropTypes.oneOf(options.overflow),
  /** To add style */
  className: PropTypes.string,
  /** Container Flexbox */
  flex: PropTypes.bool,
  /** Center Container */
  centered: PropTypes.bool,
  /** Left Aligned Container */
  leftAligned: PropTypes.bool,
  /** Right Aligned Container */
  rightAligned: PropTypes.bool,
  /** This Boolean attribute add border to see it on Storybook */
  isVisible: PropTypes.bool,
  /** 100% width when mobile*/
  fullWidthMobile: PropTypes.bool,
  /** Add border top and bottom */
  borderVertical: PropTypes.bool,
  /** Add border left and right */
  borderHorizontal: PropTypes.bool,
  /** Min Width Max Contentt */
  maxContent: PropTypes.bool,
  minContent: PropTypes.bool,
  /** Hide for mobile resolution */
  hideMobile: PropTypes.bool,
  /** Adds negative Margin*/
  negativeMargin: PropTypes.bool,
  /** Hide for tablet resolution */
  hideTablet: PropTypes.bool,
  /** Shpw for tablet resolution */
  showTablet: PropTypes.bool,
  /** Shpw for mobile resolution */
  showMobile: PropTypes.bool,
  /** Display block for Tablet Resolution */
  blockTablet: PropTypes.bool,
  /** Adds position absolute */
  absolute: PropTypes.bool,
  /** Top position in px */
  top: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Right position in px */
  right: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Bottom position in px */
  bottom: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Left position in px */
  left: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** Makes container 100% height */
  fullHeight: PropTypes.bool,
}

Container.defaultProps = {}

export default Container
