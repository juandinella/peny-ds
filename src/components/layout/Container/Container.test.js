import React from 'react'

import { cleanup, render } from '@testing-library/react'

import Container from './Container'

import initStoryshots from '@storybook/addon-storyshots'

initStoryshots({
  storyKindRegex: /^Layout\/Container$/,
})

const componentProps = {
  id: 'container',
  children: <span data-testid="container_span">Im a span</span>,
  className: 'custom_class',
}

afterEach(cleanup)

describe('<Container />', () => {
  test('should render component correctly', () => {
    const { queryByTestId } = render(<Container {...componentProps} />)

    const container = queryByTestId(componentProps.id)

    expect(container).toBeTruthy()
  })

  test('should render children correctly', () => {
    const { queryByTestId } = render(<Container {...componentProps} />)

    expect(queryByTestId(componentProps.id).children).toHaveLength(1)

    expect(queryByTestId('container_span')).toBeTruthy()

    expect(queryByTestId('container_span').textContent).toEqual('Im a span')
  })

  test('should render class correctly', () => {
    const { queryByTestId, rerender } = render(
      <Container {...componentProps} />
    )

    const container = queryByTestId(componentProps.id)

    expect(container.className).toContain('custom_class')

    rerender(<Container {...componentProps} className={null} />)

    expect(container.className).not.toContain('custom_class')
  })
})
