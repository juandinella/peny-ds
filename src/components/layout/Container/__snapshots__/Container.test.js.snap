// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Layout/Container Align Content 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 hdvqjv"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 ehDTEb"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jvLvcr"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 cuJYBn"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bolOKz"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 hGCJjf"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 hdvqjv"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Align Items 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 aikND"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 cGdGLf"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fvNXOz"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jNBmaP"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jymBn"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Border Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 jYCXcc"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jlfFIJ"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 gbsVtO"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fnVEav"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 deaiUE"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 cqCRBl"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 edzMKa"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 dqcvqH"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bghakQ"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 cDhinc"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 lhmrIR"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 ESPUj"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 hJxkzt"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Border Radiuses 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 hTIivS"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 gcjsRP"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fyDhxu"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 kcleyB"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Direction 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 kMDWSL"
    direction="row"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 koWBgW"
    direction="row-reverse"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 hQIRxT"
    direction="column"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 kGqxYm"
    direction="column-reverse"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Justify Content 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 bFLlEb"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 PYbSP"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 dMDdSj"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fZpthT"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bbhEuz"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fupJIX"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 iONyDL"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container M Align Content 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 oVIUk"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 eaUXoA"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 ibcvTU"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fqituc"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bmZmI"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 kMpYJM"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 oVIUk"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container M Align Items 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 jMHeeY"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 kokRkQ"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bJemvE"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 lhZLuI"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bRvpGY"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container M Direction 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 dpIXgw"
    direction="row"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fubpFx"
    direction="row-reverse"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 goqjyY"
    direction="column"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 wGzwh"
    direction="column-reverse"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container M Justify Content 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 hPPQDE"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 iQZhLo"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 glWIRM"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jQrbWs"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 kdfaqI"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bQOKdg"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 kAnTsQ"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container M Spaces 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 gOGjJD"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 eHkzMb"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 iNQuFD"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bUnkWP"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bXFTkX"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jkiQdT"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fUDQyj"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jQzArv"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Spaces 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 gOGjJD"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 eHkzMb"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 iNQuFD"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bUnkWP"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 bXFTkX"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jkiQdT"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fUDQyj"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 jQzArv"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
  >
    This is a Child!
  </div>
</div>
`;

exports[`Storyshots Layout/Container Wrap 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="sc-kazc2j-0 jLQQwj"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
    wrap="nowrap"
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 kElGTC"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
    wrap="wrap"
  >
    This is a Child!
  </div>
  <div
    className="sc-kazc2j-0 fLJWin"
    style={
      Object {
        "bottom": undefined,
        "left": undefined,
        "right": undefined,
        "top": undefined,
      }
    }
    wrap="wrap-reverse"
  >
    This is a Child!
  </div>
</div>
`;
